src/webapp/components/edit-sharing/EditSharing.tsx

```diff
@@ -0,0 +1,139 @@
+                .catch(error => {
+                    snackbar.error(error.message);
+                    loading.hide();
+                });
+        },
+        [compositionRoot.dataSets.save, dataSets, loading, snackbar, setDataSets, setSharingValue]
+    );
+
+    const searchSharing = React.useCallback(
+        async (value: string) => {
+            return compositionRoot.sharing.search
+                .execute(value)
+                .toPromise()
+                .then((response): SearchResult => {
+                    return {
+                        userGroups: response.userGroupAccesses.map(group => {
+                            return { displayName: group.name, id: group.id };
+                        }),
+                        users: response.userAccesses.map(user => {
+                            return { displayName: user.name, id: user.id };
+                        }),
+                    };
+                });
+        },
+        [compositionRoot.sharing.search]
+    );
+
+    const sharingMetaValue = {
+        object: {
+            id: dataSet?.id || "",
+            displayName: multipleDataSets
+                ? `[${dataSetIds.length}] ${joinDataSetsShortNames}`
+                : dataSet?.name,
+            externalAccess: false,
+            publicAccess:
+                dataSet && !sharingValue?.publicAccess
+                    ? DataSet.generateFullPermission(dataSet)
+                    : sharingValue?.publicAccess,
+            userAccesses: buildAccessByType(multipleDataSets, dataSet, "users", sharingValue),
+            userGroupAccesses: buildAccessByType(multipleDataSets, dataSet, "groups", sharingValue),
+        },
+        meta: { allowExternalAccess: false, allowPublicAccess: true },
+    };
+
+    return (
+        <ConfirmationDialog
+            cancelText={i18n.t("Close")}
+            fullWidth
+            onCancel={onCancel}
+            open
+            title={i18n.t("Sharing Settings")}
+        >
+            <SharingModal
+                meta={sharingMetaValue}
+                onChange={saveDataSets}
+                onSearch={searchSharing}
+                showOptions={sharingOptions}
+            />
+        </ConfirmationDialog>
+    );
+});
+
+function buildAccessByType(
+    multipleDataSets: boolean,
+    dataSet: Maybe<DataSet>,
+    type: AccessType,
+    sharingValue: Maybe<ShareUpdate>
+): Maybe<SharingRule[]> {
+    const selectedValues =
+        type === "users" ? sharingValue?.userAccesses : sharingValue?.userGroupAccesses;
+
+    const dataSetAccess = _(dataSet?.access || [])
+        .filter(access => access.type === type)
+        .map((access): SharingRule => {
+            return { access: access.value, displayName: access.name, id: access.id };
+        })
+        .value();
+
+    return multipleDataSets
+        ? selectedValues
+        : _([...dataSetAccess, ...(selectedValues || [])])
+              .uniqBy(access => access.id)
+              .value();
+}
```

Comment:
This logic looks like it's not view-only related, so it can be moved to the domain